### PR TITLE
support Embedded Struct. and exclude method parameter.

### PR DESCRIPTION
**Pull request by kazu**
current master ignore creation of embedded sturct member.
this PR support this.

and 

added parameter  for   target method  without creantion  .( -e option )

### CODE REVIEW

File: ifacemaker.go [hunk 1]
```diff
@@ -5,45 +5,68 @@ import (
...
+	Output         string   `cli:"o,output" usage:"Output file name. If not provided, result will be printed to stdout."`
 }
 
 func run(args *cmdlineArgs) {
 	allMethods := []string{}
 	allImports := []string{}
 	mset := make(map[string]struct{})
 	iset := make(map[string]struct{})
+	parseds := make(map[string]*maker.Parsed)
+	fmt.Printf("files=%s\n", args.Files)
```
Comment (vburenin) [inline ifacemaker.go]: is a debug print output? it should not be here.

File: ifacemaker.go [hunk 2]
```diff
@@ -5,45 +5,68 @@ import (
 	"io/ioutil"
 	"log"
 
+	"github.com/kazu/ifacemaker/maker"
```
Comment (vburenin) [inline ifacemaker.go]: It seems like you refer your own repo here. for pull request it should be reverted back to original.

File: maker/maker.go [hunk 3]
```diff
@@ -16,6 +16,12 @@ type Method struct {
 	Docs []string
 }
 
+type Parsed struct {
```
Comment (vburenin) [inline maker/maker.go]: I guess Parsed names is a little confusing. StructData sounds a bit better to me.

File: maker/maker.go [hunk 4]
```diff
@@ -37,7 +43,35 @@ func GetReceiverTypeName(src []byte, fl interface{}) (string, *ast.FuncDecl) {
 		st = st[1:]
 	}
 	return st, fd
+}
+func GetStructName(src []byte, dec interface{}) string {
+
+	if gd, ok := dec.(*ast.GenDecl); ok {
```
Comment (vburenin) [inline maker/maker.go]: it is too many ifs inside ifs, for such type of construction is better to keep code a bit flatter as this:
```Go
gd, ok := dec.(*ast.GenDecl)
if !ok || gd.Tok != token.Type {
    return ""
}
ts, ok := gd.Specs[0].(*ast.TypeSpec)
if !ok {
    return ""
}
return ts.Name.Name
```

You code is correct, but having it a bit more narrow should generally be easier to interpret.

File: maker/maker.go [hunk 5]
```diff
@@ -37,7 +43,35 @@ func GetReceiverTypeName(src []byte, fl interface{}) (string, *ast.FuncDecl) {
...
+				return ts.Name.Name
+			}
+		}
+	}
+	return ""
+}
 
+func GetEmbedded(src []byte, dec interface{}) (embeds []string) {
+
+	if gd, ok := dec.(*ast.GenDecl); ok {
```
Comment (vburenin) [inline maker/maker.go]: same here.

File: maker/maker.go [hunk 6]
```diff
@@ -114,32 +165,58 @@ func ParseStruct(src []byte, structName string, copyDocs bool) (methods []Method
...
 			}
-			params, _ := GetParameters(src, fd.Type.Params)
-			ret, merged := GetParameters(src, fd.Type.Results)
-
-			var retValues string
-			if merged {
-				retValues = fmt.Sprintf("(%s)", strings.Join(ret, ", "))
-			} else {
-				retValues = strings.Join(ret, ", ")
+			if StringSlice(exs).Contain(methodName) {
```
Comment (vburenin) [inline maker/maker.go]: Why do you make a copy every time here for exs array? Each such lookup is O(n), did you consider using a map for deduplication purposes?

Could you also name exs differently, it is not clear what it means from the function signature standpoint.

File: ifacemaker.go [hunk 7]
```diff
@@ -5,45 +5,68 @@ import (
...
 		if err != nil {
 			log.Fatal(err.Error())
 		}
-		methods, imports := maker.ParseStruct(src, args.StructType, args.CopyDocs)
-		for _, m := range methods {
-			if _, ok := mset[m.Code]; !ok {
-				allMethods = append(allMethods, m.Lines()...)
-				mset[m.Code] = struct{}{}
-			}
+		//parseds := maker.ParseStruct(src, args.StructType, args.CopyDocs)
```
Comment (vburenin) [inline ifacemaker.go]: please remove commented line.

File: ifacemaker.go (hunk 1 above)
Comment (kazu) [reply ifacemaker.go]: sorry. i will remove

File: ifacemaker.go (hunk 2 above)
Comment (kazu) [reply ifacemaker.go]: ok . fix later

File: maker/maker.go (hunk 6 above)
Comment (kazu) [reply maker/maker.go]: sorry. i dont think  becaose this is tiny and quick-hacks

**Comment by kazu** [comment]
@vburenin  thanks review. i will fix for your comment.

**Pull request closed without merging**